examples/research_projects/wav2vec2/run_common_voice.py

```diff
@@ -391,11 +394,18 @@ def extract_all_chars(batch):
     # Preprocessing the datasets.
     # We need to read the aduio files as arrays and tokenize the targets.
     def speech_file_to_array_fn(batch):
-        speech_array, sampling_rate = torchaudio.load(batch["path"])
-        batch["speech"] = resampler(speech_array).squeeze().numpy()
+        try:
+            speech_array, sampling_rate = sf.read(batch["path"] + ".wav")
+        except:
+            speech_array, sampling_rate = torchaudio.load(batch["path"])
+            speech_array = resampler(speech_array).squeeze().numpy()
+            sf.write(batch["path"] + ".wav", speech_array, sampling_rate, subtype='PCM_24')
```

Comment:
I'm not in favor of writing the processed file to the disk. When applying `.map(...)` the output of this function is already cached

Reply:
Yeah, but in my experiments it's not working after changing parameters, like filtering etc.
Sometimes I use only 10% after breaking changes for testing purposes and when using the full dataset it's not cached anymore and needs to be resampled again.
Maybe as optional argument to if audio should be stored on disk ?

Reply:
Puuh still not really in favor of it. I don't think a 10% speed-up is worth it here to write both the map function and the soundfiles to cache, but ok to do it with a flag! Let's default  the flag to False then :-)